### ISSUE TITLE
[BUG] Song Data Disappeared?

### ISSUE THREAD

**Issue by Roxy-Pop** [squiggythings/WaveTracker#48]
The short version is that I was working on a song and tried to press Ctrl+Z to undo, fat-fingered some other keys by mistake, and now all of the patterns for the song seem like they've been completely deleted from the module.

I had left the program open overnight while the computer was on sleep mode, and this happened about five minutes into returning to the project this morning. I had just created a new instrument and replacing some already placed notes with the new instrument. Like I said, I made a mistake and went to press undo, but I'm not sure what keys I pressed, because instead of undoing my last action, suddenly all of my patterns were gone. Not just from the Frame window, which had been reduced to a single pattern 00, but also the data for every channel in every pattern is now blank. The Frame and Pattern windows seem like they were completely reverted to a new file. The waves, instruments, and song settings stuck around.

I closed the program without saving and tried to reopen the file, hoping the data would still be there, but the patterns remained blank, much to my panic and confusion. I may have fat-fingered Ctrl+S somewhere in the process and accidentally saved the reset version. The other songs in the module are still just fine from what I can tell, but losing an entire track has me feeling pretty discouraged.

I am running version 1.0.3 on Windows 10.

**Comment by Dannode36** [comment]
On the other hand, making the autosaves accessible from inside WaveTracker is something I think needs to happen. I didn't even realise autosaves were a thing until I searched through the code

**Comment by squiggythings** [comment]
Autosaves aren't in 1.0.3 unfortunately, that's in the next version which is still under development. This is a pretty puzzling bug because I'm not sure how to even begin reproducing it to test, but autosaves in 1.0.4 will hopefully alleviate a lot of the trouble if anything catastrophic should happen.

**Comment by Dannode36** [comment]
Ah my apologies I got the dates of the release and commit mixed up (COVID brain fog is something else).

Totally weird bug though. As far as I'm aware, WaveTracker doesn't even have any function to wipe song data right? At least not without prompting the user to save.

**Comment by Roxy-Pop** [comment]
yeah, i initially wasn't sure if i should even post the bug report because i wasn't able to reproduce it, but i figured it would be better to put it on peoples' radar! i tried pressing every combination of keys i could think of in the vicinity of ctrl+z and nothing seemed to do it. very happy to hear about the autosave feature though, since i never remember to save as often as i should. :P
